lib/settings_yaml.rb

```diff
@@ -19,40 +19,75 @@
 #       enabled: true
 #
 # @example `config/initializers/00_settings.rb`
-#   require 'handsome_extensions/settings'
+#   require 'settings_yaml'
 #
-#   Settings = HandsomeExtensions::Settings.new.load!
+#   Settings = SettingsYaml.load!
 #
 # @example Accessing settings
 #   if Settings.cloud.enabled
 #     ...
 #   end
-class SettingsYaml < Hashie::Mash
-  # @return [Hasie::Mash]
-  def load!
-    self.deep_merge! settings_from_file('settings.yml')
-    self.deep_merge! settings_from_file('settings.local.yml')
-    self
-  end
+class SettingsYaml
+
+  APPLICATION_SETTINGS_ROOT_NAME = 'application'
+
+  class << self
+    # @param filename [String]
+    # @param environment [String]
+    # @return [Hasie::Mash]
+    def load!(filename='settings', environment=nil)
+      main_settings, local_settings  = FileParser.parse("#{filename}.yml"), FileParser.parse("#{filename}.local.yml")
+
+      result = Hashie::Mash.new
+      result.deep_merge!( application_settings(main_settings) )
+            .deep_merge!( environment_settings(main_settings, environment) )
+            .deep_merge!( application_settings(local_settings) )
+            .deep_merge!( environment_settings(local_settings, environment) )
 
-  private
+      result
+    end
+
+    private
 
-  # @param filename [String]
-  # @return [Hash]
-  def settings_from_file(filename)
-    file_path       = Rails.root.join 'config', filename
-    file_contents   = File.read(file_path)
-    parsed_file     = ERB.new(file_contents).result
-    loaded_settings = YAML.load(parsed_file)
+    # @param loaded_settings [Hash]
+    # @param environment [String]
+    # @return [Hash]
+    def environment_settings(loaded_settings, environment)
+      loaded_settings[(environment || default_env)] || {}
+    end
 
-    {}.tap do |settings|
-      if loaded_settings
-        settings.deep_merge! loaded_settings['global'] || {}
-        settings.deep_merge! loaded_settings[Rails.env] || {}
+    # @param loaded_settings [Hash]
+    # @return [Hash]
+    def application_settings(loaded_settings)
+      loaded_settings[APPLICATION_SETTINGS_ROOT_NAME] || {}
+    end
+
+    # @return [String]
+    def default_env
+      defined? Rails ? Rails.env : nil
```

Comment:
If gem supports Rails-less environment, then it would be better to use `activesupport-core-ext` gem and not the whole `activesupport`.